### PR TITLE
Added useSignInEmail hook and store universal email

### PR DESCRIPTION
**Pull request by corbanbrook**

### CODE REVIEW

**Review by yigiterdev — DISMISSED** [review]

File: examples/react/src/components/Connected.tsx [hunk 1]
```diff
@@ -41,6 +42,9 @@ export const Connected = () => {
   const { openERC1155SaleContractPaymentModal } = useERC1155SaleContractPaymentModal()
   const { data: walletClient } = useWalletClient()
   const storage = useStorage()
+  const email = useSignInEmail()
+
+  console.log('MY EMAIL', email)
```
Comment (SamueleA) [inline examples/react/src/components/Connected.tsx:47]: Can we remove this log prior to merging?

**Review by SamueleA — CHANGES_REQUESTED** [review]

File: packages/kit/src/hooks/useSignInEmail.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import { LocalStorageKey } from '../constants/localStorage'
+
+export const useSignInEmail = () => {
+  const { storage } = useConfig()
+  const { isConnected } = useAccount()
+  const [email, setEmail] = useState<null | string>(null)
+
+  const storeEmail = async () => {
+    const storedEmail = await storage?.getItem(LocalStorageKey.WaasSignInEmail)
```
Comment (SamueleA) [inline packages/kit/src/hooks/useSignInEmail.ts:14]: Do we want to rename the LocalStorageKey to remove the term "waas" from it and make it more general?

File: examples/react/src/components/Connected.tsx (hunk 1 above)
Comment (corbanbrook) [reply examples/react/src/components/Connected.tsx]: done

**Review by SamueleA — APPROVED** [review]

File: packages/kit/src/hooks/useSignInEmail.ts (hunk 2 above)
Comment (corbanbrook) [reply packages/kit/src/hooks/useSignInEmail.ts:14]: We will leave it for now and migrate it another time.

**Pull request merged**